components/director/internal/domain/apptemplate/service_test.go

```diff
@@ -154,6 +155,94 @@ func TestService_Create(t *testing.T) {
+
+	uidSvcFn := func() *automock.UIDService {
+		uidSvc := &automock.UIDService{}
+		uidSvc.On("Generate").Return(testID)
+		return uidSvc
+	}
+
+	modelAppTemplate := fixModelApplicationTemplate(testID, testName, []*model.Webhook{})
+
+	testCases := []struct {
+		Name              string
+		Input             *model.ApplicationTemplateInput
+		AppTemplateRepoFn func() *automock.ApplicationTemplateRepository
+		WebhookRepoFn     func() *automock.WebhookRepository
+		LabelUpsertSvcFn  func() *automock.LabelUpsertService
+		ExpectedError     error
+		ExpectedOutput    string
+	}{
+		{
+			Name:  "Success",
+			Input: fixModelAppTemplateInput(testName, appInputJSONString),
+			AppTemplateRepoFn: func() *automock.ApplicationTemplateRepository {
+				appTemplateRepo := &automock.ApplicationTemplateRepository{}
+				appTemplateRepo.On("Create", ctx, *modelAppTemplate).Return(nil).Once()
+				return appTemplateRepo
+			},
+			WebhookRepoFn: func() *automock.WebhookRepository {
+				webhookRepo := &automock.WebhookRepository{}
+				webhookRepo.On("CreateMany", ctx, "", []*model.Webhook{}).Return(nil).Once()
+				return webhookRepo
+			},
+			LabelUpsertSvcFn: func() *automock.LabelUpsertService {
+				labelUpsertService := &automock.LabelUpsertService{}
+				labelUpsertService.On("UpsertMultipleLabels", ctx, "", model.AppTemplateLabelableObject, "foo", map[string]interface{}{"createWithLabels": "OK", "test": "test"}).Return(nil).Once()
+				return labelUpsertService
+			},
+			ExpectedError:  nil,
+			ExpectedOutput: testID,
+		},
+		{
+			Name:  "Error when creating application template",
+			Input: fixModelAppTemplateInput(testName, appInputJSONString),
+			AppTemplateRepoFn: func() *automock.ApplicationTemplateRepository {
+				appTemplateRepo := &automock.ApplicationTemplateRepository{}
+				appTemplateRepo.On("Create", ctx, *modelAppTemplate).Return(testError).Once()
+				return appTemplateRepo
+			},
+			WebhookRepoFn: func() *automock.WebhookRepository {
+				return &automock.WebhookRepository{}
+			},
+			LabelUpsertSvcFn: func() *automock.LabelUpsertService {
+				return &automock.LabelUpsertService{}
+			},
+			ExpectedError:  testError,
+			ExpectedOutput: "",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			appTemplateRepo := testCase.AppTemplateRepoFn()
+			webhookRepo := testCase.WebhookRepoFn()
+			labelUpsertSvc := testCase.LabelUpsertSvcFn()
+			idSvc := uidSvcFn()
+			svc := apptemplate.NewService(appTemplateRepo, webhookRepo, idSvc, labelUpsertSvc, nil)
+
+			// WHEN
+			result, err := svc.CreateWithLabels(ctx, *testCase.Input, map[string]interface{}{"createWithLabels": "OK"})
+
+			// THEN
+			if testCase.ExpectedError != nil {
+				require.Error(t, err)
+				assert.Contains(t, err.Error(), testCase.ExpectedError.Error())
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, testCase.ExpectedOutput, result)
+
+			appTemplateRepo.AssertExpectations(t)
```

Comment:
You can move these statements at line 226 and use _defer mock.AssertExpectationsForObjects()_ instead.